oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/EventHookCallback.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.alarm.provider;
+
+import org.apache.skywalking.apm.network.event.v3.Event;
+import org.apache.skywalking.apm.network.event.v3.Source;
+import org.apache.skywalking.apm.network.event.v3.Type;
+import org.apache.skywalking.oap.server.analyzer.event.EventAnalyzerModule;
+import org.apache.skywalking.oap.server.analyzer.event.EventAnalyzerService;
+import org.apache.skywalking.oap.server.core.alarm.AlarmCallback;
+import org.apache.skywalking.oap.server.core.alarm.AlarmMessage;
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.source.DefaultScopeDefine;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * EventCallBack: When an alert is present, an event is generated for each alert message. These events are then sent to the internal event analyzer.
+ *
+ */
+public class EventHookCallback implements AlarmCallback {
+
+    private final ModuleManager manager;
+
+    public EventHookCallback(ModuleManager manager) {
+        this.manager = manager;
+    }
+
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessage) {
+        if (Objects.isNull(this.manager)) {
+            return ;
+        }
+        EventAnalyzerService analyzerService = manager.find(EventAnalyzerModule.NAME).provider().getService(EventAnalyzerService.class);
+        alarmMessage.forEach(a -> {
+            for (Event event : constructCurrentEvent(a)) {
+                if (Objects.nonNull(event)) {
+                    analyzerService.analyze(event);
+                }
+            }
+        });
+    }
+
+    private Event[] constructCurrentEvent(AlarmMessage msg) {
+        Event[] events = new Event[2];
+        long now = System.currentTimeMillis();
+        Event.Builder builder = Event.newBuilder()
+                .setUuid(UUID.randomUUID().toString())
+                .setName("Alarm")
```

Comment:
@kezhenxu94 Do we suppose to have a document to maintain the knowing event name? As this seems going to keep adding.

Reply:
> I want to check about the time range, could you show me a screenshot at minute levels? And the alarm event should cover several minutes?

![image](https://user-images.githubusercontent.com/36784140/115374981-ee333d80-a1ff-11eb-93c9-cbc90c7c08fe.png)
![image](https://user-images.githubusercontent.com/36784140/115375017-f55a4b80-a1ff-11eb-82f0-1f99b9305d53.png)

Reply:
@chenmudu Thanks, this looks good. Please follow the other comments.

Reply:
> @chenmudu Thanks, this looks good. Please follow the other comments.

Yeah, Copy.

Reply:
> @kezhenxu94 Do we suppose to have a document to maintain the knowing event name? As this seems going to keep adding.

Yes. We can add a section `Known Events` in the event.md. @chenmudu

Reply:
Yeah, Copy that.